### ISSUE TITLE
Add a keyboard switching option

### ISSUE THREAD

**Issue by PhilC813** [FossifyOrg/Keyboard#76]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Keyboard/issues) or [closed](https://github.com/FossifyOrg/Keyboard/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Keyboard/discussions) or [closed](https://github.com/FossifyOrg/Keyboard/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Keyboard/blob/main/CONTRIBUTING.md).
- [X] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Provide a way to quickly change to another installed keyboard app rather than requiring the user to make the change at the Android settings level.

Google Gboard and FUTO Keyboard have this capability when holding the spacebar. I'm not sure if this is the best way to implement it or not.

### Why do you want this feature?

This feature would improve usability by allowing users to switch to other keyboards when needed, as each keyboard has its own strengths and weaknesses. This is especially useful given that Fossify Keyboard is still in early development.

**Comment by luewolf** [comment]
Android 13 provides this out of the box. A keyboard icon is automatically displayed next to the back, home and running apps keys if multiple keyboards are enabled. Clicking on it allows selecting another keyboard. Might be interesting for people using Android <13 though. I don't know how it's handled in Android 14.

![android_keyboard_switch_key](https://github.com/user-attachments/assets/c46eaf9d-48db-4a6d-82b0-45da0d6ffc39)

**Comment by Philippe-Choquette** [comment]
@luewolf That's a good point. However, it only applies if you use system navigation with the buttons at the bottom of the screen. My device is on Android 14 and by default, it uses gestures for navigation; OEMs probably prefer this option because it maximizes screen real estate.

![Screenshot_20240915_141908.jpg](https://github.com/user-attachments/assets/21bb1a43-94f4-49db-8b9f-08d57bf7db70)

**Comment by luewolf** [comment]
Good point. It would make sense to add an optional key that can be enabled/disabled in the settings, then.

@Philippe-Choquette where on the keyboard layout would you place the keyboard switching key?

**Comment by Philippe-Choquette** [comment]
I'm no UI expert, but if we compare with other keyboards ~



**Gboard**
Holding the spacebar invokes the system keyboard change menu.

![Screenshot_20240917_142832.jpg](https://github.com/user-attachments/assets/4ae135cf-6f64-4a3a-9b63-ff883a67b5f3)

_Gboard has an advantage against other keyboards because it adds input languages directly to the system keyboards, so it doesn't require a proprietary language switching menu._



**FUTO Keyboard**
Holding the spacebar invokes its proprietary language switching menu. You can then tap "Switch keyboard" to pick another keyboard.

![Screenshot_20240917_142758.jpg](https://github.com/user-attachments/assets/d3dc43fe-aa1d-44aa-8211-0a61ff1f0acf)

You can also add an optional language switching button to the top bar, but it's only designed to switch between the languages on FUTO Keyboard.



Fossify Keyboard already has a proprietary language switching menu invoked when holding the emoji key. So the easiest implementation would be to follow the FUTO Keyboard implementation and add a "Switch keyboard" button to that menu. There could be 10 different ways to implement this tho, and I don't know which would be best.

**Comment by naveensingh** [comment]
> adds input languages directly to the system keyboards

Those are called [IME subtypes](https://developer.android.com/develop/ui/views/touch-and-input/creating-input-method#IMESubTypes). I think Gboard does it best but it's not as quick-n-easy as just-add-another-dialog option.

**Comment by Mafiadoener36** [comment]
Sorry for duplicate - searched different keywords! 

Please dont hide it behind switch keyboards button behind change language button - inefficient - just follow most other keyboards - longpress space for switching - for having unified design and nothing exotic sticking out.
